release/release.go

```diff
@@ -111,36 +110,59 @@ var fileExists = func(path string) bool {
 	return err == nil
 }
 
-var filesystemRootType = func() (int64, error) {
-	var statfs syscall.Statfs_t
-	if err := syscall.Statfs("/", &statfs); err != nil {
-		return 0, fmt.Errorf("cannot statfs filesystem root")
+// filesystemRootType returns the filesystem type mounted at "/".
+var filesystemRootType = func() (string, error) {
+	// We scan /proc/mounts, which contains space-separated values:
+	// [irrelevant] [mount point] [fstype] [irrelevant...]
+	// Here are some examples on some platforms:
+	// WSL1       :  rootfs / wslfs rw,noatime 0 0
+	// WSL2       :  /dev/sdc / ext4 rw,relatime,discard,errors=remount-ro,data=ordered 0 0
+	// lxc on WSL2:  /dev/loop0 / btrfs rw,relatime,idmapped,space_cache,user_subvol_rm_allowed,subvolid=259,subvol=/containers/testlxd 0 0
+	// We search for mount point = "/", and return the fstype.
+	//
+	// This should be done by osutil.LoadMountInfo but that would cause a dependency cycle
+	file, err := os.Open("/proc/mounts")
```

Comment:
If we do `var procMountPath = "/proc/mounts"` and have a `MockProcMountPath` this becomes easier to test without having to look at the real /proc/mounts in the test itself.

Reply:
Agreed. Changed.